pkg/admissioncontroller/webhooks/admission/auditpolicy/admission.go

```diff
@@ -0,0 +1,285 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auditpolicy
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"time"
+
+	acadmission "github.com/gardener/gardener/pkg/admissioncontroller/webhooks/admission"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/controllermanager/controller/shoot"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/version"
+
+	"github.com/go-logr/logr"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	audit_internal "k8s.io/apiserver/pkg/apis/audit"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	auditv1alpha1 "k8s.io/apiserver/pkg/apis/audit/v1alpha1"
+	auditv1beta1 "k8s.io/apiserver/pkg/apis/audit/v1beta1"
+	auditvalidation "k8s.io/apiserver/pkg/apis/audit/validation"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	// HandlerName is the name of this admission webhook handler.
+	HandlerName = "auditpolicy_validator"
+	// WebhookPath is the HTTP handler path for this admission webhook handler.
+	WebhookPath = "/webhooks/validate-audit-policies"
+
+	auditPolicyConfigMapDataKey = "policy"
+)
+
+var (
+	runtimeScheme = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(runtimeScheme)
+	decoder       = codecs.UniversalDecoder()
+	_             = auditv1alpha1.AddToScheme(runtimeScheme)
+	_             = auditv1beta1.AddToScheme(runtimeScheme)
+	_             = auditv1.AddToScheme(runtimeScheme)
+	_             = audit_internal.AddToScheme(runtimeScheme)
+
+	shootGVK     = metav1.GroupVersionKind{Group: "core.gardener.cloud", Kind: "Shoot", Version: "v1beta1"}
+	configmapGVK = metav1.GroupVersionKind{Group: "", Kind: "ConfigMap", Version: "v1"}
+)
+
+// New creates a new handler for validating audit policies.
+func New(logger logr.Logger) *handler {
+	return &handler{
+		logger: logger,
+	}
+}
+
+type handler struct {
+	apiReader client.Reader
+	decoder   *admission.Decoder
+	logger    logr.Logger
+}
+
+var _ admission.Handler = &handler{}
+
+func (h *handler) InjectAPIReader(reader client.Reader) error {
+	h.apiReader = reader
+	return nil
+}
+
+func (h *handler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+// Handle implements the webhook handler for audit log policy validation
+func (h *handler) Handle(ctx context.Context, request admission.Request) admission.Response {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	switch request.Kind {
+	case shootGVK:
```

Comment:
This should also catch requests to `shoots.core.gardener.cloud/v1alpha1` and basically all possible future Gardener API versions.

Reply:
Done also for configmap